File: src/p0997/golang/solution.go
package main

func findJudge(n int, trusts [][]int) int {
	degree := make([]int, n + 1)
	for _, trust := range trusts {
		degree[trust[0]]--
		degree[trust[1]]++
	}
	for i := 1; i <= n; i++ {
		if degree[i] == n - 1 {
			return i
		}
	}
	return -1
}

File: src/p0997/cpp/solution.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int findJudge(int N, vector<vector<int>> &trusts) {
        vector<int> degree(N + 1, 0);
        for (auto &trust : trusts) {
            degree[trust[0]]--;
            degree[trust[1]]++;
        }
        for (int i = 1; i <= N; i++) {
            if (degree[i] == N - 1) {
                return i;
            }
        }
        return -1;
    }
};
